example/lib/src/ui/device_list.dart

```diff
@@ -70,19 +68,24 @@ class _DeviceListState extends State<DeviceList> {
       log("Starting to listen for devices...");
 
       _clearDeviceList();
-      _scanSubscription =
-          _ble.scanForDevices(withService: uuid).listen((device) {
-        if (!_devices.any((d) => d.id == device.id)) {
-          log("New device found: $device");
-
+      setState(() {
+        _scanSubscription =
+            _ble.scanForDevices(withService: uuid).listen((device) {
           if (mounted) {
             setState(() {
-              _devices.add(device);
+              final knownDeviceIndex =
+                  _devices.indexWhere((d) => d.id == device.id);
+
+              if (knownDeviceIndex >= 0) {
+                _devices[knownDeviceIndex] = device;
+              } else {
+                log("New device found: $device");
```

Comment:
Honestly, I'd kill all the logging in the example app.

Reply:
kill it

Reply:
As discussed via a side-channel, let's kill it during the course of #35.